pkg/graph/resolvers/tag.go

```diff
@@ -0,0 +1,148 @@
+	switch tagType {
+	case models.TagTypeUserAccount:
+		return UserAccountGetByIDLOADER(ctx, *EntityUUID)
+	case models.TagTypePossibleSolution:
+		return PossibleOperationalSolutionGetByID(logger, store, *EntityIntID)
+	default:
+		return nil, fmt.Errorf(" no tagged entity configured for table: %s", tagType)
+	}
+
+}
+
+// UpdateTaggedHTMLMentionsAndRawContent updates the tagged html with the correct entity ids, and updates the RAW HTMl with the new representation of the mentions
+func UpdateTaggedHTMLMentionsAndRawContent(ctx context.Context, store *storage.Store, tHTML *models.TaggedHTMLInput, getAccountInformation userhelpers.GetAccountInfoFunc) error {
+	errs := []error{}
+	for _, mention := range tHTML.Mentions {
+		if mention.EntityDB != nil && mention.EntityDB != "" { // Check if the id is set, if not do logic to get the entity record created in the db / return the entity needed
+			continue
+		}
+		tagType := mention.Type
+
+		// Conditionally set the entity DB by tag type
+		switch tagType {
+		case models.TagTypeUserAccount:
+			isMacUser := false
+			collabAccount, err := userhelpers.GetOrCreateUserAccount(ctx, store, mention.EntityRaw, false, isMacUser, getAccountInformation)
+			if err != nil {
+				errs = append(errs, err)
+				continue
+			}
+			mention.EntityUUID = &collabAccount.ID
+			mention.EntityDB = mention.EntityUUID
+
+		case models.TagTypePossibleSolution:
+			logger := appcontext.ZLogger(ctx)
+
+			sol, err := store.PossibleOperationalSolutionGetByKey(logger, models.OperationalSolutionKey(mention.EntityRaw))
+			if err != nil {
+				errs = append(errs, err)
+				continue
+			}
+			mention.EntityIntID = &sol.ID
+			mention.EntityDB = mention.EntityIntID
+		default:
+			return fmt.Errorf("could not set entity id because the tag type is invalid %s", tagType)
+		}
+
+		// Updated the parent Raw content with the new fields
+		newHTML, err := mention.ToHTML()
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		// Update the Tagged HTML Raw content by replacing the tags old value, with the new representation
+		newTotalRaw := strings.Replace(string(tHTML.RawContent), string(mention.RawHTML), string(newHTML), -1)
+		tHTML.RawContent = models.HTML(newTotalRaw)
+
+		mention.RawHTML = newHTML
+	}
+	if len(errs) > 0 {
+		return fmt.Errorf("issues encountered getting database ids for tagged entities. %v", errs) // We aren't wrapping these errors because this is an array
+	}
+	return nil
+}
+
+// TagCollectionCreate converts an array of mentions, and creates an array in the database for unique tags.
+func TagCollectionCreate(logger *zap.Logger, store *storage.Store, principal authentication.Principal,
+	taggedField string, taggedTable string, taggedContentID uuid.UUID, mentions []*models.HTMLMention) ([]*models.Tag, error) {
+
+	tags := models.TagArrayFromHTMLMentions(taggedField, taggedTable, taggedContentID, mentions)
+
+	uniqTags := lo.UniqBy(tags, func(tag *models.Tag) string {
```

Comment:
I'd like opinions on this please. This prevents an entity being tagged in the same table twice. There might be multiple mentions referring to the entity, but they would only have one tag. I think this will be useful for in app notifications as they come up